allensdk/internal/brain_observatory/time_sync.py

```diff
@@ -18,37 +18,43 @@
 LONG_STIM_THRESHOLD = 0.2     # seconds
 MAX_MONITOR_DELAY = 0.07     # seconds
 
-VERSION_1_KEYS = {
-    "photodiode": "stim_photodiode",
-    "2p": "2p_vsync",
-    "stimulus": "stim_vsync",
-    "eye_camera": "cam2_exposure",
-    "behavior_camera": "cam1_exposure",
-    "acquiring": "2p_acquiring",
-    "lick_sensor": "lick_1"
-    }
-
-# MPE is changing keys. This isn't versioned in the file.
-VERSION_2_KEYS = {
-    "photodiode": "photodiode",
-    "2p": "2p_vsync",
-    "stimulus": "stim_vsync",
-    "eye_camera": "eye_tracking",
-    "behavior_camera": "behavior_monitoring",
-    "acquiring": "2p_acquiring",
-    "lick_sensor": "lick_sensor"
-    }
-
 
 def get_keys(sync_dset):
     """Get the correct lookup for line labels.
 
     This method is fragile, but not all old data contains the full list
     of keys.
     """
-    if "cam2_exposure" in sync_dset.line_labels:
-        return VERSION_1_KEYS
-    return VERSION_2_KEYS
+    key_dict = {
+            "photodiode": ["stim_photodiode", "photodiode"],
+            "2p": "2p_vsync",
+            "stimulus": "stim_vsync",
+            "eye_camera": ["cam2_exposure", "eye_tracking"],
+            "behavior_camera": ["cam1_exposure", "behavior_monitoring"],
+            "acquiring": "2p_acquiring",
+            "lick_sensor": ["lick_1", "lick_sensor"]
+            }
+    label_set = set(sync_dset.line_labels)
+    for key, value in key_dict.items():
+        if isinstance(value, list):
+            diff = label_set - set(value)
+            print(diff)
+            if value[1] in label_set:
```

Comment:
assumes a list of at least 2. the set operation way was agnostic to list size,

Reply:
It doesn't return the correct key value pairs, len(diff) is never ==1 in set operation way as the sync_set holds all the values for the different keys in it's set. I need to think about the implementation of the set a bit

Reply:
```
In [29]: possible = set(['a', 'b'])                                                

In [30]: longer_list = set(['b', 'c', 'd', 'e'])                                   

In [31]: longer_list - possible                                                    
Out[31]: {'c', 'd', 'e'}

In [32]: possible - longer_list                                                    
Out[32]: {'a'}
```

Reply:
Removed this in the latest update, moved to set functions.

Reply:
I ended up using possible.intersect(longer_list) as the above has the output {'a', 'b'} after set subtraction

Reply:
@djkapner should resolve this conversation